lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -20,6 +20,7 @@ def extract_facts_from_request(request)
       raise ArgumentError, _("Facts but no fact format provided for %{request}") % { request: request.key }
     end
 
+    facts = nil
```

Comment:
I confused myself reading this at first thinking it was unnecessary because the next big if expression was exhaustive. I think it would relieve my confusion if that expression were extracted into its own function, then you could just assign the result to facts directly.